xbrowse_server/templates/family/family_home.html

```diff
@@ -633,18 +658,122 @@ <h4 class="modal-title">PhenoTips</h4>
 		}
 	}
 	if (numMatchesFound>0){ 
-		$('#mmeMatchSummaryTbl').DataTable();
+		$(resultTable).DataTable();
 		//this is to readjust the column width after unhiding
-		var datTbl = $('#mmeMatchSummaryTbl').DataTable();
-	    $("#mmeMatchSummaryTbl").css("width","100%")
+		var datTbl = $(resultTable).DataTable();
+	    $(resultTable).css("width","100%")
 	    datTbl.columns.adjust().draw(); 
-	    $('#mmeResultsSummaryContainer').show();
+	    $(resultContainer).show();
 	}
 	return numMatchesFound;
 }
 
 
 
+/**
+ * Show results summary in the specified table and container
+ */
+function renderOpenMMESourceMatches(seqrId,matches,projectId, resultTable,resultContainer){	
+
+	console.log(matches);
+	for (var result_origin in matches['match_results']){
+		//map of id to full patient obj
+		if (200 == matches['match_results'][result_origin]['status_code']){
+			for (var i=0; i<matches['match_results'][result_origin]['result']['results'].length; i++){
+				var singlePatientResult;
+				singlePatientResult =  matches['match_results'][result_origin]['result']['results'][i];
+		     	var patient=singlePatientResult['patient'];
+		     	var score=singlePatientResult['score']['patient'];
+		     	if (score == null){
+		     		score="";
+		     	}
+
+						     	var html = '<tr>';
+						     	
+					     		html += '<td>';
+					     		html += seqrId;
+					     		html += '</td>';
+
+					     		html += '<td>';
+					     		if (patient['id'].match('OMIM') != null){
+						     		html += '<a target="_blank" href="https://www.omim.org/entry/' + patient['id'].replace('OMIM:','') + '">' + patient['id'] + '</a>';
+					     		}
+					     		else{
+					     			html += patient['id'];
+					     		}
+					     		html += '</td>';	
+					     		
+					     		html += '<td>';
+					     		html += patient['label'];
+					     		html += '</td>';	
+
+					     		html += '<td>';
+					     		html += score;
+					     		html += '</td>';
+						     				     		     						     		
+					     		//GenomicFeatures
+					     		html += '<td>';
+					     		if (patient.hasOwnProperty('genomicFeatures') &&  patient['genomicFeatures'].length>0){
+						     		for (var k=0; k<patient['genomicFeatures'].length;k++){
+						     			
+						     			var geneId=patient['genomicFeatures'][k]['gene']['id'];
+						     			if (geneId.search('ENSG')==0){
+						     				var asyncGeneId='asyncGene_' + result_origin + '_'+ i.toString() + '_' + k.toString();
+						     				html += '<div id="'+ asyncGeneId + '"></div>';
+						     				addHgncToEnsembl(geneId, asyncGeneId);
```

Comment:
Great to see gene names being added. 
Later on, would be nice if this was done server-side using info from the database, to avoid N+1 http requests.

Reply:
Agreed! I think in this case the code was using an existing endpoint that multiple functions used. I think the newer additions are using server-side and we can adjust these next time around for sure